flake.nix

```
@@ -80,7 +80,12 @@
               ];
             };
 
-            nixos = base: custom base.config.system.build.toplevel "$PROFILE/bin/switch-to-configuration switch";
+            nixos = base: custom base.config.system.build.toplevel ''
+              $PROFILE/bin/switch-to-configuration switch
+
+              # https://github.com/serokell/deploy-rs/issues/31
+              sed -i '/^default /d' ${base.config.boot.loader.efi.efiSysMountPoint}/loader/loader.conf || :
```

I would check if the EFI loader is enabled, otherwise every non systemd-boot host will get `sed: can't read /boot/loader/loader.conf: No such file or directory` in their deploy logs.

---

We could also just pipe stderr/stdout to /dev/null

---

I'd rather see the unnecessary spam than accidentally suppress a useful error.

---

So, I think actually checking if `config.boot.loader.systemd-boot.enable == true` is the best solution here. Something like

```suggestion
              ${with base.config.boot.loader;
              lib.optionalAttrs systemd-boot.enable
              "sed -i '/^default /d' ${efi.efiSysMountPoint}/loader/loader.conf"}
```

---

I've added that change now